client/app/containers/tests-runs/tests-runs-search/modal/search-modal.controller.js

```diff
@@ -0,0 +1,180 @@
+'use strict'
+
+const SearchModalController = function SearchModalController(onApply, onReset, testsRunsService, windowWidthService, DEFAULT_SC, $rootScope, TestRunService, ProjectService, $q, FilterService, $mdDateRangePicker, $timeout, $mdDialog) {
+    'ngInject';
+    const subjectName = 'TEST_RUN';
+    const SELECT_CRITERIAS = ['ENV', 'PLATFORM', 'PROJECT', 'STATUS'];
+    const STATUSES = ['PASSED', 'FAILED', 'SKIPPED', 'ABORTED', 'IN_PROGRESS', 'QUEUED', 'UNKNOWN'];
+    const vm = {
+        isMobile: windowWidthService.isMobile,
+        statuses: STATUSES,
+        selectedRange: {
+            selectedTemplate: null,
+            selectedTemplateName: null,
+            dateStart: null,
+            dateEnd: null,
+            showTemplate: false,
+            fullscreen: false
+        },
+        searchParams: testsRunsService.getLastSearchParams(),
+        onChangeSearchCriteria,
+        openDatePicker,
+        closeModal,
+        onModalReset,
+        onReset: onReset,
+        onApply: onApply,
+        onModalApply
+    };
+
+    vm.$onInit = init;
+
+    return vm;
+
+    function init() {
+        loadFilters();
```

Comment:
Parent controller has all needed data, you should pass them to this modal instead of loading once again